src/map/mod.rs

```diff
@@ -0,0 +1,315 @@
+
+pub struct MapViewState<DS, CS> {
+    map_dom_state: DS,
+    children_state: CS,
+    map_ctx: MapCtx,
+}
+
+#[derive(Debug, Clone)]
+pub enum MapMessage {
+    InitMap,
+    ZoomEnd(f64), // TODO: remove this
+}
+
+/// Distinctive ID for better debugging
+const MAP_VIEW_ID: ViewId = ViewId::new(1236068);
+
+impl<State, Action, Children> View<State, Action, ViewCtx, DynMessage>
+    for Map<State, Action, Children>
+where
+    State: 'static,
+    Action: 'static,
+    Children: MapChildren<State>,
+{
+    type Element = <MapDomView<State, Action> as View<State, Action, ViewCtx, DynMessage>>::Element;
+
+    type ViewState = MapViewState<
+        <MapDomView<State, Action> as View<State, Action, ViewCtx, DynMessage>>::ViewState,
+        Children::SeqState,
+    >;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        ctx.with_id(MAP_VIEW_ID, |ctx| {
+            let (map_dom_element, map_dom_state) = self.map_view.build(ctx);
+
+            let map_options = leaflet::MapOptions::default();
+            let leaflet_map = leaflet::Map::new_with_element(&map_dom_element.node, &map_options);
+
+            let mut elements = AppendVec::default();
+            let mut map_ctx = MapCtx::new(leaflet_map.clone(), ctx.message_thunk());
+            let children_state = self.children.seq_build(&mut map_ctx, &mut elements);
+
+            let view_state = MapViewState {
+                map_ctx,
+                map_dom_state,
+                children_state,
+            };
+
+            // Is the following an issue?
+            // Does it require on being in the DOM tree?
+            // Can the tile-layer be added before running `apply_zoom_and_center`?
+
+            // We have to postpone the map initiation
+            // because the DOM element has been created at this point in time
+            // but has not yet been mounted.
+            let thunk = ctx.message_thunk();
+            spawn_local(async move {
+                thunk.push_message(MapMessage::InitMap);
+            });
+            (map_dom_element, view_state)
+        })
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        log::debug!("Rebuild map");
+        ctx.with_id(MAP_VIEW_ID, |ctx| {
+            let element =
+                self.map_view
+                    .rebuild(&prev.map_view, &mut view_state.map_dom_state, ctx, element);
+            if prev.zoom != self.zoom || prev.center != self.center {
+                apply_zoom_and_center(&view_state.map_ctx.map, self.zoom, self.center);
+            }
+            let mut splice = MapChildrenSplice;
+            log::debug!("seq_rebuild children");
+            self.children.seq_rebuild(
+                &prev.children,
+                &mut view_state.children_state,
+                &mut view_state.map_ctx,
```

Comment:
This also needs an updated view_path from the parent when not going the named lifetime route. because the `id_path` could have changed dynamically, e.g. when a parent view sequence has changed its structure